dlp/src/main/java/com/example/dlp/DeId.java

```diff
@@ -0,0 +1,237 @@
+/**
+ * Copyright 2017 Google Inc.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.dlp;
+
+import com.google.cloud.dlp.v2beta1.DlpServiceClient;
+import com.google.common.io.BaseEncoding;
+import com.google.privacy.dlp.v2beta1.*;
+import com.google.privacy.dlp.v2beta1.InfoTypeTransformations.InfoTypeTransformation;
+import com.google.privacy.dlp.v2beta1.CryptoReplaceFfxFpeConfig.FfxCommonNativeAlphabet;
+import com.google.protobuf.ByteString;
+import org.apache.commons.cli.*;
+
+public class DeId {
+
+  private static void deidentifyWithMask(String string, Character maskingCharacter, int numberToMask) {
+    // [START dlp_deidentify_mask]
+    // instantiate a client
+    try (DlpServiceClient dlpServiceClient = DlpServiceClient.create()) {
+
+      // The string to deidentify
+      // string = "My SSN is 372819127";
+
+      // (Optional) The maximum number of sensitive characters to mask in a match
+      // If omitted from the request or set to 0, the API will mask any matching characters
+      // numberToMask = 5;
+
+      // (Optional) The character to mask matching sensitive data with
+      // maskingCharacter = 'x';
+
+      ContentItem contentItem =
+          ContentItem.newBuilder()
+              .setType("text/plain")
+              .setValue(string)
+              .build();
+
+      CharacterMaskConfig characterMaskConfig =
+          CharacterMaskConfig.newBuilder()
+              .setMaskingCharacter(maskingCharacter.toString())
+              .setNumberToMask(numberToMask)
+              .build();
+
+      PrimitiveTransformation primitiveTransformation =
+          PrimitiveTransformation.newBuilder()
+              .setCharacterMaskConfig(characterMaskConfig)
+              .build();
+
+      InfoTypeTransformation infoTypeTransformationObject =
+          InfoTypeTransformation.newBuilder()
+              .setPrimitiveTransformation(primitiveTransformation)
+              .build();
+
+      InfoTypeTransformations infoTypeTransformationArray =
```

Comment:
Given all these objects that get created just to create a request, it would be to provide a comment above each to state what it is. It is unfortunate that `InfoTypeTransformations` object exists to just provide an `InfoTypeTransformation[]` surface.  `PrimitiveTransformation`, `InfoTypeTransformation` also seem pretty confusing. In an ideal world, `PrimitveTransformation`, `InfoTypeTransformation`, `InfoTypeTransformations` are un-necessary objects.

Reply:
I had similar comments with the Node client libraries, to be honest. I'll follow up with the API/product team(s) about this.

(TODO, so I can CTRL-F for this comment.)